generator/generator.md

```diff
@@ -0,0 +1,246 @@
+            en: 'Yes'
+        }
+    },
+    {
+        value: 'non',
+        label: {
+            fr: 'Non',
+            en: 'No'
+        }
+    }
+];
+```
+
+### Choices Example
+
+![Choices example](./assets/images/choicesExample.png)
+
+## Generate InputRange
+
+In Excel, the `InputRange` tab generates sliders in `inputRange.tsx`. The provided table example will generate the following TypeScript code, defining the `confiantInputRange` object in `inputRange.tsx`.
+
+### InputRange Fields
+
+| Field          | Description                                     | Example                 | Type        |
+|----------------|-------------------------------------------------|-------------------------|-------------|
+| inputRangeName | Name of the input range                         | confiantInputRange      | string      |
+| labelFrMin     | French label for the minimum value              | Pas du tout confiant    | string      |
+| labelFrMax     | French label for the maximum value              | Très confiant           | string      |
+| labelEnMin     | English label for the minimum value             | Not at all confident    | string      |
+| labelEnMax     | English label for the maximum value             | Very confident          | string      |
+| minValue       | Minimum numerical value for the input range     | -10                     | number      |
+| maxValue       | Maximum numerical value for the input range     | 100                     | number      |
+| unitFr         | Unit in French                                  | %                       | string      |
+| unitEn         | Unit in English                                 | %                       | string      |
+
+### InputRange TypeScript
+
+```typescript
+// inputRange.tsx
+export const confiantInputRange = {
+    labels: [
+        {
+            fr: 'Pas du tout confiant',
+            en: 'Not at all confident'
+        },
+        {
+            fr: 'Très confiant',
+            en: 'Very confident'
+        }
+    ],
+    minValue: -10,
+    maxValue: 100,
+    formatLabel: (value, language) => {
+        return value + ' ' + (language === 'fr' ? '%' : '%');
+    }
+};
+```
+### InputRange Example
+![InputRange example](./assets/images/inputRangeExample.png)
+
+## Generate Libelles
+
+In the context of your survey logic, libelles (labels) play a crucial role in presenting questions to respondents in different languages. This Excel table below outlines the fields in `Widgets` tab used to define libelles, along with an example and the expected output in a `introduction.yml` file.
+
+### Libelles Fields
+
+| Field       | Description          | Example                                                           | Type   |
+|-------------|----------------------|-------------------------------------------------------------------|--------|
+| path        | Path to the question | introduction.Q1_quelleOrganisation                                | string |
```

Comment:
For the labels, you can look at the fillLocales script, which is already in Evolution, and re-use its logic. It allows to add contexts to the labels, like _cati for telephone interviews vs web, or _one for first person vs nickname formulations, with column headers of the type [lang][_context] (eg. fr, fr_cati, fr_one, fr_cati_one) for 4 different labels in french depending on context.

Reply:
Yes, the generateLibelles.py is pretty much the same file as fillLocales, but I did make some changes to make it work with all the other Generator scripts.

Reply:
I will add this to the todo list.